### PR TITLE
caf: ble_smp: Update binary file name

### PR DESCRIPTION
**Pull request by MarekPieta**
Change updates information about the binary file that should be used for the DFU procedure. This is needed to align documentation with use case where MCUboot bootloader in direct-xip mode is used.

Change also updates references to ble_smp_transfer_event.

Jira: NCSDK-21651

### CODE REVIEW

**Review by kapi-no — COMMENTED** [review]
Remember about nRF desktop `smp.rst` documentation file

**Comment by MarekPieta** [comment]
> Remember about the alignment of the nRF desktop `smp.rst` documentation file

The nRF Desktop specific file does not mention the exact file name. It also links to the CAF module doc which explains further details. I think it's fine and there is no need to duplicate information about the file name.

**Review by annwoj — APPROVED** [review]

File: doc/nrf/libraries/caf/ble_smp.rst [hunk 1]
```diff
@@ -38,19 +38,27 @@ The |smp| supports registering OS management handlers automatically, which you c
...
-
-You can perform the DFU using for example the `nRF Connect for Mobile`_ application.
+You can perform the DFU using, for example, the `nRF Connect for Mobile`_ application.
 The :guilabel:`DFU` button appears in the tab with the connected Bluetooth devices.
 After pressing the button, you can select the :file:`*.bin` file that is to be used for the firmware update.
+Depending on the selected image slot either the :file:`zephyr/app_update.bin` or :file:`zephyr/mcuboot_secondary_app_update.bin` file must be used for the background DFU procedure.
+
+If the MCUboot bootloader uses the built-in direct-xip mode, you must upload the image to the slot that is currently unused.
+If the MCUboot bootloader uses the built-in swap mode instead, you must use the :file:`zephyr/app_update.bin` file.
+The :file:`zephyr/app_update.bin` file is the application image that can be booted from the primary slot.
```
Comment (mkapala-nordic) [inline doc/nrf/libraries/caf/ble_smp.rst:57]: maybe it should be explained which file is for which slot right after line `"Depending on the selected image slot (...)"` and mention there `mcuboot_secondary_app_update.bin`

File: applications/nrf_desktop/doc/smp.rst [hunk 2]
```diff
@@ -13,6 +13,7 @@ The |smp| is responsible for performing a Device Firmware Upgrade (DFU) over Blu
 You can perform the DFU using for example the `nRF Connect for Mobile`_ application.
 The :guilabel:`DFU` button appears in the tab with the connected Bluetooth® devices.
 After pressing the button, you can select the :file:`*.bin` file that is to be used for the firmware update.
+See the :ref:`nrf_desktop_image_transfer_over_smp` section for information about which binary file should be used for the DFU image transfer.
```
Comment (kapi-no) [inline applications/nrf_desktop/doc/smp.rst:16]: Shouldn't we add this sentence to the dfu_mcumgr.rst file?

Comment (MarekPieta) [reply applications/nrf_desktop/doc/smp.rst:16]: Will do under task that applies remaining review comments to the file

File: applications/nrf_desktop/README.rst [hunk 3]
```diff
@@ -1930,12 +1930,21 @@ Depending on the side on which the process is handled:
...
+.. _nrf_desktop_image_transfer_over_smp:
+
 Image transfer over SMP
 ~~~~~~~~~~~~~~~~~~~~~~~
 
 If the MCUboot bootloader is selected, the update image can also be transferred in the background through the :ref:`nrf_desktop_ble_smp`.
 The `nRF Connect for Mobile`_ application uses binary files for the image transfer over the Simple Management Protocol (SMP).
-Depending on the selected image slot either the :file:`zephyr/app_update.bin` or :file:`zephyr/mcuboot_secondary_app_update.bin` file must be used for the background DFU procedure.
+The :guilabel:`DFU` button appears in the tab of the connected Bluetooth device that supports the image transfer over the Simple Management Protocol (SMP).
+After pressing the button, you can select the :file:`*.bin` file that is to be used for the firmware update.
```
Comment (annwoj) [inline applications/nrf_desktop/README.rst:1941]:
```suggestion
After pressing the button, you can select the :file:`*.bin` file used for the firmware update.
```

File: applications/nrf_desktop/README.rst [hunk 4]
```diff
@@ -1930,12 +1930,21 @@ Depending on the side on which the process is handled:
...
 
+.. _nrf_desktop_image_transfer_over_smp:
+
 Image transfer over SMP
 ~~~~~~~~~~~~~~~~~~~~~~~
 
 If the MCUboot bootloader is selected, the update image can also be transferred in the background through the :ref:`nrf_desktop_ble_smp`.
 The `nRF Connect for Mobile`_ application uses binary files for the image transfer over the Simple Management Protocol (SMP).
-Depending on the selected image slot either the :file:`zephyr/app_update.bin` or :file:`zephyr/mcuboot_secondary_app_update.bin` file must be used for the background DFU procedure.
+The :guilabel:`DFU` button appears in the tab of the connected Bluetooth device that supports the image transfer over the Simple Management Protocol (SMP).
```
Comment (annwoj) [inline applications/nrf_desktop/README.rst:1940]:
```suggestion
The :guilabel:`DFU` button appears in the tab of the connected Bluetooth device that supports the image transfer over the SMP.
```
Acronym already covered in the line above

File: applications/nrf_desktop/README.rst [hunk 5]
```diff
@@ -1930,12 +1930,21 @@ Depending on the side on which the process is handled:
...
+After pressing the button, you can select the :file:`*.bin` file that is to be used for the firmware update.
+
+The following firmware update files are generated in the build directory after building your application for configuration with the :ref:`nrf_desktop_ble_smp` enabled:
```
Comment (annwoj) [inline applications/nrf_desktop/README.rst:1943]:
```suggestion
After building your application for configuration with the :ref:`nrf_desktop_ble_smp` enabled, the following firmware update files are generated in the build directory:
```

File: doc/nrf/libraries/caf/ble_smp.rst [hunk 6]
```diff
@@ -38,19 +38,31 @@ The |smp| supports registering OS management handlers automatically, which you c
...
 
-The module periodically submits ``ble_smp_transfer_event`` while the image is being uploaded.
-
-You can perform the DFU using for example the `nRF Connect for Mobile`_ application.
-The :guilabel:`DFU` button appears in the tab with the connected Bluetooth devices.
+You can perform the DFU using, for example, the `nRF Connect for Mobile`_ application.
+The :guilabel:`DFU` button appears in the tab of the connected Bluetooth device that supports the image transfer over the Simple Management Protocol (SMP).
 After pressing the button, you can select the :file:`*.bin` file that is to be used for the firmware update.
 
+The following firmware update files are generated in the build directory after building your application for configuration with the |smp| enabled:
```
Comment (annwoj) [inline doc/nrf/libraries/caf/ble_smp.rst:54]:
```suggestion
After building  your application for configuration with the |smp| enabled, the following firmware update files are generated in the build directory:
```

**Review by kapi-no — APPROVED** [review]

**Pull request merged**